lib/src/Uncertainty/Algorithm/Sensitivity/RankSobolSensitivityAlgorithm.cxx

```diff
@@ -0,0 +1,420 @@
+  graph.setBoundingBox(Interval(lowerBound, upperBound));
+  graph.setIntegerXTick(true);
+  graph.setLegendPosition("topright");
+  return graph;
+}
+
+/** Method that draw (plot) the sensitivity graph */
+Graph RankSobolSensitivityAlgorithm::draw() const
+{
+  Graph graph(DrawSobolFirstIndices(inputDescription_, getAggregatedFirstOrderIndices()));
+  if (outputDesign_.getDimension() > 1)
+    graph.setTitle(OSS() << "Aggregated Sobol' indices - " << getClassName());
+  else
+    graph.setTitle(OSS() << "Sobol' indices - " << getClassName());
+  const UnsignedInteger dimension = aggregatedFirstOrderIndices_.getDimension();
+
+  // Draw confidence intervals
+  if (confidenceLevel_ > 0.0)
+  {
+    const Interval foInterval(getFirstOrderIndicesInterval());
+
+    // transform data
+    Sample data(2, 2);
+    for (UnsignedInteger k = 0; k < dimension; ++k)
+    {
+      // Relative to FirstOrder
+      data(0, 0) = (k + 1);
+      data(0, 1) = foInterval.getLowerBound()[k];
+      data(1, 0) = (k + 1);
+      data(1, 1) = foInterval.getUpperBound()[k];
+      graph.add(Curve(data, "red", "solid", 2, ""));
+    }
+  }
+  return graph;
+}
+
+
+
+// Function that computes aggregated indices  using Vi + variance
+// Indices are aggregated with respect to output dimensions
+Point RankSobolSensitivityAlgorithm::computeAggregatedIndices(const Sample & Vi,
+    const Point & variance) const
+{
+  // Generic implementation
+  const UnsignedInteger inputDimension = Vi.getDimension();
+  const UnsignedInteger outputDimension = Vi.getSize();
+  if (inputDimension == 1)
+  {
+    return Point(Vi[0]);
+  }
+
+  // Compute sum of Var(Y^k)
+  Scalar sumVariance = variance.norm1();
+
+  // Compute aggregated indices
+  return Point(Vi.computeMean() * (outputDimension / sumVariance));
+}
+
+
+
+/** void method that computes confidence interval */
+void RankSobolSensitivityAlgorithm::computeBootstrapDistribution() const
+{
+  // Build interval using sample variance
+  // Mean reference is the Sensitivity values
+  const Point FirstOrder(getFirstOrderIndices());
+
+  if (bootstrapSize_ > 0)
+  {
+    // Temporary samples that stores the first/total indices
+    const UnsignedInteger inputDimension = inputDescription_.getSize();
+    Sample bsFO(0, inputDimension);
+
+    UnsignedInteger bootstrapSampleSize(ResourceMap::GetAsScalar("RankSobolSensitivityAlgorithm-DefaultBootstrapSampleRatio")*size_);
+        
+    const UnsignedInteger blockSize = std::min(bootstrapSize_, ResourceMap::GetAsUnsignedInteger("SobolIndicesAlgorithm-DefaultBlockSize"));
+    const UnsignedInteger maximumOuterSampling = bootstrapSize_ / blockSize;
+    const UnsignedInteger modulo = bootstrapSize_ % blockSize;
+    const UnsignedInteger lastBlockSize = modulo == 0 ? blockSize : modulo;
+
+    // Use of KPermutations to perform bootstrap without replication
+    Distribution KPermutation = KPermutationsDistribution(bootstrapSampleSize, size_);
```

Comment:
KPermutationsDistribution kPermutation(bootstrapSampleSize, size);

Reply:
not sure to understand why you proposed this change. "size" should be the global size of the sample, so the new attribute size_.

Reply:
yes but size_ should not be an attribute as its redundant, you can use a local variable instead

Reply:
drop the size_ attribute, use a local variable "UnsignedInteger size = inputDesign_.getSize()" when needed

Reply:
This attribute is inherited from the mother class.

Reply:
oh ok, but you should still drop it from your .hxx (and save/load methods)